sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/http/policy/ArmChallengeAuthenticationPolicy.java

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.experimental.http.policy;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.core.experimental.implementation.AuthenticationChallenge;
+import com.azure.core.experimental.implementation.AzureEnvironment;
+import com.azure.core.experimental.implementation.ARMScopeHelper;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.CoreUtils;
+import reactor.core.publisher.Mono;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * The pipeline policy that applies a token credential to an HTTP request
+ * with "Bearer" scheme in ARM challenge based authentication scenarios
+ * .
+ * This is a temporary class to support ARM Challenge based authentication. It will
+ * move to azure-resource-manager package.
+ */
+public class ArmChallengeAuthenticationPolicy extends BearerTokenAuthenticationChallengePolicy {
+    private static final Pattern AUTHENTICATION_CHALLENGE_PATTERN =
+        Pattern.compile("(\\w+) ((?:\\w+=\".*?\"(?:, )?)+)(?:, )?");
+    private static final Pattern AUTHENTICATION_CHALLENGE_PARAMS_PATTERN =
+        Pattern.compile("(?:(\\w+)=\"([^\"\"]*)\")+");
+    private static final String CLAIMS_PARAMETER = "claims";
+    private final String[] scopes;
+    private final AzureEnvironment environment;
+    private static final String ARM_SCOPES_KEY = "ARMScopes";
+
+    /**
+     * Creates ArmChallengeAuthenticationPolicy.
+     *
+     * @param credential the token credential to authenticate the request
+     * @param environment the environment with endpoints for authentication
+     * @param scopes the scopes used in credential, using default scopes when empty
+     */
+    public ArmChallengeAuthenticationPolicy(TokenCredential credential,
+                                            AzureEnvironment environment, String... scopes) {
+        super(credential);
+        this.scopes = scopes;
+        this.environment = environment;
+    }
+
+
+    @Override
+    public Mono<Void> onBeforeRequest(HttpPipelineCallContext context) {
+        return Mono.defer(() -> {
+            TokenRequestContext trc;
```

Comment:
Do we need a local variable for this? It appears to only be passed into `authorizeRequest`